### PR TITLE
Bambora Online ePay 3DS support

### PR DESCRIPTION
**Pull request by AllanWLie**
Adds 3DS support to the epay payment method.
---- Unit Test ----
Loaded suite test/unit/gateways/epay_test.rb
16 tests, 55 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
---- Remote Test ----
Loaded suite test/remote/gateways/remote_epay_test.rb
9 tests, 15 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/epay.rb [hunk 1]
```diff
@@ -158,6 +160,16 @@ def add_instant_capture(post, option)
         post[:instantcapture] = option ? 1 : 0
       end
 
+      def add_3ds_auth(post, options)
+        if options[:three_d_secure]
+          post[:eci] = options.dig(:three_d_secure, :eci)
+          post[:xid] = options.dig(:three_d_secure, :xid)
+          post[:cavv] = options.dig(:three_d_secure, :cavv)
+          post[:threeds_version] = options.dig(:three_d_secure, :version)
+          post[:ds_transaction_id] = options.dig(:three_d_secure, :ds_transaction_id)
```
Comment (filipebarcos) [inline lib/active_merchant/billing/gateways/epay.rb]: In many cases `xid` and `ds_transaction_id` are exclusive. Would it be ok to have one of these values being `nil`?

File: lib/active_merchant/billing/gateways/epay.rb [hunk 2]
```diff
@@ -268,7 +279,7 @@ def xml_builder(params, soap_call)
 
       def authorize_post_data(params = {})
         params[:language] = '2'
-        params[:cms] = 'activemerchant'
+        params[:cms] = 'activemerchant_3ds'
```
Comment (filipebarcos) [inline lib/active_merchant/billing/gateways/epay.rb]: can we get the reasoning behind this change?

Comment (filipebarcos) [reply lib/active_merchant/billing/gateways/epay.rb]: Also, any chance we could support something like `application_id || 'activemerchant'`?https://github.com/activemerchant/active_merchant/blob/341b53eb0b3a129808389c5b4f7714b6a80d06cd/lib/active_merchant/billing/gateways/checkout_v2.rb#L98

File: test/remote/gateways/remote_epay_test.rb [hunk 3]
```diff
@@ -5,12 +5,10 @@ def setup
     Base.mode = :test
 
     @gateway = EpayGateway.new(fixtures(:epay))
-
     @credit_card = credit_card('3333333333333000')
     @credit_card_declined = credit_card('3333333333333102')
-
     @amount = 100
-    @options = {order_id: '1'}
+    @options = {order_id: '1', three_d_secure: { eci: '7', xid: '123', cavv: '456', version: '2', ds_transaction_id: '798' }}
```
Comment (filipebarcos) [inline test/remote/gateways/remote_epay_test.rb]: we mostly never have xid and ds_transaction_id on the same request (same as before)

Comment (AllanWLie) [reply test/remote/gateways/remote_epay_test.rb]: eci, cavv and xid are required for 3ds-V1. The ds_transaction_id is only required when 3DS-V2 kicks in. We can handle empty values.

File: lib/active_merchant/billing/gateways/epay.rb (hunk 1 above)
Comment (AllanWLie) [reply lib/active_merchant/billing/gateways/epay.rb]: Yes.

File: lib/active_merchant/billing/gateways/epay.rb (hunk 2 above)
Comment (filipebarcos) [reply lib/active_merchant/billing/gateways/epay.rb]: `application_id` could be set by a user of Active Merchant to identify their platform. With a static value all requests using active merchant are grouped on the same bucket. If you had support for `application_id`, you could differentiate between "shopping cart" A and B

**Comment by filipebarcos** [comment]
@AllanWLie you go approvals and unless you're planning to change anything I'll go ahead and merge these changes

**Pull request merged**